Assets/MRTK/Services/CameraSystem/ProjectionOverride.cs

```diff
@@ -0,0 +1,121 @@
+        /// Coroutine function to set the camera matrices back to their defaults
+        /// </summary>
+        /// <returns>Coroutine enumerator</returns>
+        IEnumerator ResetViewMatricesOnFrameEnd()
+        {
+            yield return new WaitForEndOfFrame();
+            Camera cam = GetComponent<Camera>();
+            cam.ResetStereoViewMatrices();
+            cam.ResetStereoProjectionMatrices();
+        }
+
+        /// <inheritdoc />
+        void OnPreCull()
+        {
+            if (!ReadingModeEnabled)
+            {
+                return;
+            }
+
+            StartCoroutine(ResetViewMatricesOnFrameEnd());
+
+            IntPtr nativeStruct = UnityEngine.XR.XRDevice.GetNativePtr();
+            if (nativeStruct != IntPtr.Zero)
+            {
+                HolographicFrameNativeData s = System.Runtime.InteropServices.Marshal.PtrToStructure<HolographicFrameNativeData>(nativeStruct);
+
+                if (s.IHolographicFramePtr != IntPtr.Zero)
+                {
+                    var holographicCamera = (HolographicCamera)System.Runtime.InteropServices.Marshal.GetObjectForIUnknown(s.IHolographicCameraPtr);
+                    var holographicFrame = (HolographicFrame)System.Runtime.InteropServices.Marshal.GetObjectForIUnknown(s.IHolographicFramePtr);
+                    HolographicFramePrediction prediction = holographicFrame.CurrentPrediction;
+
+                    if (holographicCamera.CanOverrideViewport)
+                    {
+                        for (int i = 0; i < prediction.CameraPoses.Count; ++i)
+                        {
+                            HolographicCameraPose cameraPose = prediction.CameraPoses[i];
+
+                            // Default spatial locator is the HMD
+                            SpatialLocator defaultSpatialLocator = SpatialLocator.GetDefault();
+
+                            SpatialLocatorAttachedFrameOfReference attachedFrameOfReference = defaultSpatialLocator.CreateAttachedFrameOfReferenceAtCurrentHeading();
+                            SpatialCoordinateSystem attachedCoordinateFrame = attachedFrameOfReference.GetStationaryCoordinateSystemAtTimestamp(prediction.Timestamp);
+
+                            HolographicStereoTransform stereoProjection = cameraPose.ProjectionTransform;
+                            float ResolutionScale = 45.0f / 33.0f;
+
+
+                            Matrix4x4 leftProj =  CameraCache.Main.GetStereoProjectionMatrix(Camera.StereoscopicEye.Left);
+                            Matrix4x4 rightProj =  CameraCache.Main.GetStereoProjectionMatrix(Camera.StereoscopicEye.Right);
+                            leftProj.m00 *= ResolutionScale;
+                            leftProj.m11 *= ResolutionScale;
+                            rightProj.m00 *= ResolutionScale;
+                            rightProj.m11 *= ResolutionScale;
+                            CameraCache.Main.SetStereoProjectionMatrix(Camera.StereoscopicEye.Left, leftProj);
```

Comment:
Also less important, but it looks like we'll be setting this on `CameraCache.Main` multiple times if `prediction.CameraPoses.Count` is greater than 1.